flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -301,10 +301,10 @@ public boolean databaseExists(String databaseName) throws CatalogException {
 	}
 
 	@Override
-	public void dropDatabase(String name, boolean ignoreIfNotExists) throws DatabaseNotExistException,
-			DatabaseNotEmptyException, CatalogException {
+	public void dropDatabase(String name, boolean ignoreIfNotExists, boolean restrict)
```

Comment:
how about match hive client API by changing the param to `cascade` rather than `restrict`? and we don't need to change the param in client.dropDatabase(name, true, ignoreIfNotExists, **!restrict**);

Reply:
But if so, we must call `client.dropDatabse(name, true, ignoreifNotExists, !cascade)`.
The `!cascade` is not consistent with the original method.

Reply:
`cascade` is more consistent with the client API? https://hive.apache.org/javadocs/r3.1.2/api/org/apache/hadoop/hive/metastore/IMetaStoreClient.html#dropDatabase-java.lang.String-boolean-boolean-boolean-

Reply:
Yes. Addressed~